Sources/Common/DataModel/Polygon/Constants.js

```diff
@@ -6,6 +6,13 @@ export const PolygonWithPointIntersectionState = {
   FAILURE: -1,
   OUTSIDE: 0,
   INSIDE: 1,
-  INTERSECTION: 2,
-  ON_LINE: 3,
```

Comment:
can't a point be both OUTSIDE AND INSIDE, hence ON_LINE ?

Reply:
In the C++ implementation, this case returns inside the polygon. It could indeed differentiate that case however.